src/github.com/couchbase/sync_gateway/auth/principal.go

```diff
@@ -102,5 +102,9 @@ type User interface {
 	// to, annotated with the sequence number at which access was granted.
 	FilterToAvailableChannels(channels base.Set) ch.TimedSet
 
+	// Returns a TimedSet containing channels that the user has access to, that aren't present in the
+	// input set
+	GetAddedChannels(channels ch.TimedSet) *base.Set
```

Comment:
This is a little confusing since the comment say it's returning a TimedSet, where the code is actually returning a base.Set.  Can you update the comment?

I don't think it makes sense to pass a pointer to a base.Set back to the caller, because a base.Set is just a type alias for a map:

```
type Set map[string]present
```

and a map is already a reference type, and I can't think of any good reason to be taking a pointer to a reference type.  I've seen it mentioned as an anti-pattern and can dig up a reference if needed.